pkg/utils/managedresources/managedresources.go

```diff
@@ -186,20 +189,43 @@ func deployManagedResource(ctx context.Context, secret *builder.Secret, managedR
 	return nil
 }
 
-// Delete deletes the managed resource and its secret with the given name in the given namespace.
-func Delete(ctx context.Context, client client.Client, namespace string, name string, secretNameWithPrefix bool) error {
+// Delete deletes the managed resource and its secrets with the given name in the given namespace.
+func Delete(ctx context.Context, c client.Client, namespace string, name string, secretNameWithPrefix bool) error {
 	secretName := SecretName(name, secretNameWithPrefix)
 
-	if err := builder.NewManagedResource(client).
-		WithNamespacedName(namespace, name).
-		Delete(ctx); err != nil {
-		return fmt.Errorf("could not delete managed resource '%s/%s': %w", namespace, name, err)
+	mr := &resourcesv1alpha1.ManagedResource{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace}}
+	mrKey := client.ObjectKeyFromObject(mr)
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: namespace}}
+
+	err := c.Get(ctx, mrKey, mr)
+	if err != nil && apierrors.IsNotFound(err) {
+		// just try to delete the secret with generated name
+		if err := client.IgnoreNotFound(c.Delete(ctx, secret)); err != nil {
+			return fmt.Errorf("could not delete secret '%s' of managed resource: %w", client.ObjectKeyFromObject(secret).String(), err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("could not get managed resource '%s': %w", mrKey.String(), err)
 	}
 
-	if err := builder.NewSecret(client).
-		WithNamespacedName(namespace, secretName).
-		Delete(ctx); err != nil {
```

Comment:
Should we drop the `Delete` function since it's rather "dangerous" to rely on?

Reply:
Do you mean to drop it from the `builder/secrets.go` file? I am OK with that suggestion since its usage is error prone.

Reply:
I will also remove the whole `Secrets` type, since it is not used anywhere in g/g and honestly I do not see it as any useful API. I will also include this in the release notes as a breaking change